### PR TITLE
add nll-by-default blog post

### PR DESCRIPTION
**Pull request by nikomatsakis**
NLL is finally going to be used by default! I wanted to highlight this achievement and in particular the hard work of a bunch of people. It also gives us a chance to hype up polonius and educate people a bit about some of the limitations of the existing borrow checker.

### CODE REVIEW

**Comment by nikomatsakis** [comment]
cc @rust-lang/core

File: posts/2022-06-10-nll-by-default.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+## How does removing the old borrow checker affect users?
+
+At this point, we have almost completely merged "migration mode" and "regular mode", so switching to NLL will have very little impact on the user experience. A number of diagnostics changed, mostly for the better -- [Jack Huey gives the full details in his blost post](https://jackh726.github.io/rust/2022/06/10/nll-stabilization.html).
+
+## Credit where credit is due
+
+The work to remove the old borrow checker has been going on for years. It's been a long, tedious, and largely thankless process. We'd like to take a moment to highlight the various people involved and make sure they are recognized for their hard work:
+
+* [Jack Huey](https://github.com/jackh726/), for driving the final details of stabilization (diagnostics, reconciling differences in behavior).
```
Comment (joshtriplett) [inline posts/2022-06-10-nll-by-default.md:21]:
```suggestion
* [Jack Huey](https://github.com/jackh726/) ([sponsorship page](https://github.com/sponsors/jackh726)), for driving the final details of stabilization (diagnostics, reconciling differences in behavior).
```

File: posts/2022-06-10-nll-by-default.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+* [Aaron Hill](https://github.com/Aaron1011), for numerous improvements to bring NLL diagnostics up to par.
+* [Matthew Jasper](https://github.com/matthewjasper), for reconciling errors around higher-ranked lifetimes and other core diagnostics improvements.
+* [Rémy Rakic](https://github.com/lqd), for rebasing Matthew's PR as well as doing other independent diagnostics work.
+* [Christopher Vittal](https://github.com/chrisvittal), for removing "compare" mode (don't ask).
+* [Centril](https://github.com/centril), for doing a lot of early work reconciling migration mode and the regular mode.
+* And of course the members of the NLL working group (myself, [Felix Klock](https://github.com/pnkfelix), [Santiago Pastorino](https://github.com/spastorino), [Matthew Jasper](https://github.com/matthewjasper), [David Wood](https://github.com/davidtwco], [Rémy Rakic](https://github.com/lqd), [Paul Daniel Faria](https://github.com/nashenas88), [Nick Nethercote](https://github.com/nnethercote)) who delivered the feature in the first place.
```
Comment (joshtriplett) [inline posts/2022-06-10-nll-by-default.md:28]:
```suggestion
* And of course the members of the NLL working group (myself, [Felix Klock](https://github.com/pnkfelix), [Santiago Pastorino](https://github.com/spastorino), [Matthew Jasper](https://github.com/matthewjasper), [David Wood](https://github.com/davidtwco), [Rémy Rakic](https://github.com/lqd), [Paul Daniel Faria](https://github.com/nashenas88), [Nick Nethercote](https://github.com/nnethercote)) who delivered the feature in the first place.
```

Typo fix.

**Review by joshtriplett — APPROVED** [review]

**Review by rylev — APPROVED** [review]
The date needs to be changed but otherwise looks good!

File: posts/2022-06-10-nll-by-default.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
+---
+layout: post
+title: "Non-lexical lifetimes (NLL) fully stable"
+author: Niko Matsakis
+team: the NLL working group <https://www.rust-lang.org/governance/teams/compiler#Non-Lexical%20Lifetimes%20(NLL)%20working%20group>
+release: true
+---
+
+As of the Rust 1.63, the "non-lexical lifetimes" (NLL) work will be enabled by default. "But," I hear you saying, "wasn't NLL included in [Rust 2018]?" And yes, yes it was! But at that time, NLL was only enabled for Rust 2018 code, while Rust 2015 code ran in "migration mode". When in "migration mode," the compiler would run both the old *and* the new borrow checker and compare the results. This way, we could give warnings for older code that should never have compiled in the first place; we could also limit the impact of any bugs in the new code. Over time, we have limited migration mode to be closer and closer to just running the new-style borrow checker: in the next release, that process completes, and all Rust code will be checked with NLL.
```
[the remark this answers is not in the document]
Comment (nikomatsakis) [reply posts/2022-06-10-nll-by-default.md:9]: Good call out. I'll add a small example

File: posts/2022-06-10-nll-by-default.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+[Rust 2018]: https://blog.rust-lang.org/2018/12/06/Rust-1.31-and-rust-2018.html
+
+## How does removing the old borrow checker affect users?
+
+At this point, we have almost completely merged "migration mode" and "regular mode", so switching to NLL will have very little impact on the user experience. A number of diagnostics changed, mostly for the better -- [Jack Huey gives the full details in his blost post](https://jackh726.github.io/rust/2022/06/10/nll-stabilization.html).
```
Comment (programmerjake) [inline posts/2022-06-10-nll-by-default.md:15]:
```suggestion
At this point, we have almost completely merged "migration mode" and "regular mode", so switching to NLL will have very little impact on the user experience. A number of diagnostics changed, mostly for the better -- [Jack Huey gives the full details in his blog post](https://jackh726.github.io/rust/2022/06/10/nll-stabilization.html).
```

**Comment by nikomatsakis** [comment]
I've been holding off on updating this as we await the fate of https://github.com/rust-lang/rust/issues/98095 (fix pending in https://github.com/rust-lang/rust/pull/98109). It'd be a shame to post this triumphant blog post and then have to pull out the code :)

**Comment by Mark-Simulacrum** [comment]
@nikomatsakis or @jackh726 -- would one of you have the time to polish this post up so we can merge it sometime in the next couple weeks? We'd like to link to it from the main 1.63 release blog post, which will be posted August 11th -- I'd like to get this ready to go by August 3rd so there's some buffer in between them :)

**Comment by jackh726** [comment]
I'll polish it up later this evening.

**Comment by Mark-Simulacrum** [comment]
@jackh726 Did you get a chance to take a look here?

**Comment by jackh726** [comment]
@Mark-Simulacrum Should be good now. I added a bit of text to springboard the user to learning about NLL, if they so choose.

**Comment by Mark-Simulacrum** [comment]
Thanks! I think we can publish this tomorrow (ideally) or Friday - I'll try to remember to do that tomorrow, along with bumping the date.

**Review by Mark-Simulacrum — APPROVED** [review]

File: posts/2022-08-05-nll-by-default.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
+---
+layout: post
+title: "Non-lexical lifetimes (NLL) fully stable"
+author: Niko Matsakis
+team: the NLL working group <https://www.rust-lang.org/governance/teams/compiler#Non-Lexical%20Lifetimes%20(NLL)%20working%20group>
+release: false
+---
+
+As of the Rust 1.63, the "non-lexical lifetimes" (NLL) work will be enabled by default. NLL is the second iteration of Rust's borrow checker. The [RFC] actually does quite a nice job of highlighting some of the motivating examples. "But," I hear you saying, "wasn't NLL included in [Rust 2018]?" And yes, yes it was! But at that time, NLL was only enabled for Rust 2018 code, while Rust 2015 code ran in "migration mode". When in "migration mode," the compiler would run both the old *and* the new borrow checker and compare the results. This way, we could give warnings for older code that should never have compiled in the first place; we could also limit the impact of any bugs in the new code. Over time, we have limited migration mode to be closer and closer to just running the new-style borrow checker: in the next release, that process completes, and all Rust code will be checked with NLL.
```
Comment (Mark-Simulacrum) [inline posts/2022-08-05-nll-by-default.md:9]:
```suggestion
As of Rust 1.63 (releasing next week), the "non-lexical lifetimes" (NLL) work will be enabled by default. NLL is the second iteration of Rust's borrow checker. The [RFC] actually does quite a nice job of highlighting some of the motivating examples. "But," I hear you saying, "wasn't NLL included in [Rust 2018]?" And yes, yes it was! But at that time, NLL was only enabled for Rust 2018 code, while Rust 2015 code ran in "migration mode". When in "migration mode," the compiler would run both the old *and* the new borrow checker and compare the results. This way, we could give warnings for older code that should never have compiled in the first place; we could also limit the impact of any bugs in the new code. Over time, we have limited migration mode to be closer and closer to just running the new-style borrow checker: in the next release, that process completes, and all Rust code will be checked with NLL. 
```

**Pull request merged**